tile.go

```diff
@@ -15,15 +24,75 @@ type Tile struct {
 	Y         int
 	Lat       float64
 	Long      float64
-	Tolerance *float64
-	Extent    *float64
+	Tolerance float64
+	Extent    float64
+	Buffer    float64
+
+	// These values are cached
+	cached bool
+	// The width and height of the region.
+	xspan float64
+	yspan float64
+	// This is the computed bounding box.
+	extent  [2][2]float64
+	bufpext [2][2]float64
+}
+
+func NewTile(z, x, y int) (t *Tile) {
+	t = &Tile{
+		Z:         z,
+		X:         x,
+		Y:         y,
+		Buffer:    DefaultTileBuffer,
+		Extent:    DefaultExtent,
+		Tolerance: DefaultEpislon,
+	}
+	t.Lat, t.Long = t.Num2Deg()
+	t.Init()
+	return t
+}
+func NewTileLatLong(z int, lat, lon float64) (t *Tile) {
+	t = &Tile{
+		Z:         z,
+		Lat:       lat,
+		Long:      lon,
+		Buffer:    DefaultTileBuffer,
+		Extent:    DefaultExtent,
+		Tolerance: DefaultEpislon,
+	}
+	t.X, t.Y = t.Deg2Num()
+	t.Init()
+	return t
+}
+
+func (t *Tile) Init() {
+	max := 20037508.34
+
+	//	resolution
+	res := (max * 2) / math.Exp2(float64(t.Z))
+	t.cached = true
+	t.extent = [2][2]float64{
+		{
+			-max + (float64(t.X) * res), // MinX
+			max - (float64(t.Y) * res),  // Miny
+		},
+		{
+			-max + (float64(t.X) * res) + res, // MaxX
+			max - (float64(t.Y) * res) - res,  // MaxY
+
+		},
+	}
+	t.xspan = t.extent[1][0] - t.extent[0][0]
+	t.yspan = t.extent[1][1] - t.extent[0][1]
+	t.bufpext = [2][2]float64{{0 - t.Buffer, 0 - t.Buffer}, {t.Extent + t.Buffer, t.Extent + t.Buffer}}
+
 }
 
 func (t *Tile) Deg2Num() (x, y int) {
 	x = int(math.Floor((t.Long + 180.0) / 360.0 * (math.Exp2(float64(t.Z)))))
 	y = int(math.Floor((1.0 - math.Log(math.Tan(t.Lat*math.Pi/180.0)+1.0/math.Cos(t.Lat*math.Pi/180.0))/math.Pi) / 2.0 * (math.Exp2(float64(t.Z)))))
 
-	return
+	return x, y
```

Comment:
unnecessary since x,y are named returns.

Reply:
Yeah, but it still better to list the names. I not a fan of bare returns; as functions grow it becomes harder to know what is getting returned. I like naming returns because it gives documentation to the reader of the function what the function is returning. Especially when the type of the return is the same.